doc/authentication.md

```diff
@@ -0,0 +1,47 @@
+
+Cockpit Authentication
+================================
+
+Cockpit authorizes users by looking at the Authorization header. The contents
+of that header are sent to an external command that is then responsible to:
+
+ * verify the given credentials
+ * setup an appropriate session and enviroment based on those credentials
+ * launch a bridge that can communicate with cockpit-ws on stdin and stdout.
+
+Cockpit ships with the cockpit-session command that is able to handle basic and gssapi
+authentication. Cockpit supports per auth scheme configuration so that alternate commands
+can be used.
+
+To configure an auth scheme add a section to cockpit.conf for that scheme. For example
+
+```
+[bearer]
+command = example-verify-token
+base64-decode = false
```

Comment:
After thinking about this more (and looking at the ssh code below) I think a configuration scheme like this may be best:

```
[bearer]
action = spawn-authenticator-with-header
command = /my/path/example-verify-token

[basic]
action = ssh-connect
host = 127.0.0.1

[negotiate]
action = none
```

We might have:

 * spawn-login-with-header
 * spawn-login-with-decoded
 * remote-ssh-login
 * none

While we don't have to implement the whole scheme above right now, lets leave space for it in the configuration. What do you think?